In [21]:
import pandas as pd
import matplotlib.pyplot as plt

## First table. Number of people by religion

In [22]:
number_of_people = pd.read_csv('number-of-people-by-religion.csv')
number_of_people.head()

,Entity,Code,Year,Buddhists,Hindus,Muslims,Jews,Christians,No religion,Other religions
0,Afghanistan,AFG,2010,< 10000,< 10000,28210000,< 10000,30000,< 10000,30000
1,Afghanistan,AFG,2020,< 10000,< 10000,39020000,< 10000,< 10000,< 10000,40000
2,Africa,OWID_AFR,2010,8,134,48085,9,53658,2712,2615
3,Africa,OWID_AFR,2020,10,147,61814,6,70314,3007,2784
4,Albania,ALB,2010,< 10000,< 10000,2050000,< 10000,590000,280000,< 10000


In [23]:
number_of_people.dtypes

Entity             object
Code               object
Year                int64
Buddhists          object
Hindus             object
Muslims            object
Jews               object
Christians         object
No religion        object
Other religions    object
dtype: object

In [24]:
number_of_people.isna().sum() / len(number_of_people)

Entity             0.0
Code               0.0
Year               0.0
Buddhists          0.0
Hindus             0.0
Muslims            0.0
Jews               0.0
Christians         0.0
No religion        0.0
Other religions    0.0
dtype: float64

#### No missing values

In [25]:
print(len(number_of_people['Entity'].unique()))
number_of_people['Entity'].unique()

214


array(['Afghanistan', 'Africa', 'Albania', 'Algeria', 'Angola',
       'Argentina', 'Armenia', 'Aruba', 'Asia', 'Asia-Pacific (Pew)',
       'Australia', 'Austria', 'Azerbaijan', 'Bahamas', 'Bahrain',
       'Bangladesh', 'Barbados', 'Belarus', 'Belgium', 'Belize', 'Benin',
       'Bhutan', 'Bolivia', 'Bosnia and Herzegovina', 'Botswana',
       'Brazil', 'Brunei', 'Bulgaria', 'Burkina Faso', 'Burundi',
       'Cambodia', 'Cameroon', 'Canada', 'Cape Verde',
       'Central African Republic', 'Chad', 'Channel Islands', 'Chile',
       'China', 'Colombia', 'Comoros', 'Congo', 'Costa Rica',
       "Cote d'Ivoire", 'Croatia', 'Cuba', 'Curacao', 'Cyprus', 'Czechia',
       'Democratic Republic of Congo', 'Denmark', 'Djibouti',
       'Dominican Republic', 'East Timor', 'Ecuador', 'Egypt',
       'El Salvador', 'Equatorial Guinea', 'Eritrea', 'Estonia',
       'Eswatini', 'Ethiopia', 'Europe', 'Europe (Pew)', 'Fiji',
       'Finland', 'France', 'French Guiana', 'French Polynesia', 'Gabon',
 

The “entity” column contains the names of 214 countries
You can see them all above. 

In [26]:
number_of_people['Buddhists'].unique()

array(['< 10000', '8', '10', '10000', '33567', '31543', '336320000',
       '316110000', '600000', '670000', '20000', '30000', '950000',
       '1010000', '520000', '580000', '250000', '220000', '14060000',
       '16240000', '380000', '660000', '76100000', '53380000', '40000',
       '202', '255', '2010000', '2540000', '330000', '470000', '260000',
       '290000', '840000', '630000', '8690000', '9550000', '1780000',
       '1860000', '370000', '53580000', '46990000', '4050000', '4710000',
       '340000', '110000', '130000', '5680000', '6400000', '60000',
       '90000', '1460000', '1680000', '42960000', '47210000', '2470000',
       '2380000', '50000', '70000', '402', '510', '3960000', '5050000',
       '400000', '68', '74', '1730000', '28', '27', '12470000', '9850000',
       '14640000', '15700000', '4850000', '4540000', '64090000',
       '67620000', '150000', '280000', '310000', '3580000', '4380000',
       '24130000', '22580000', '342750000', '324190000'], dtype=object)

We have a problem in that the columns containing the number of people of a given faith may contain both very small values and “< 10,000” 
The question is what to do with them: 
1) Replace them with something around 10,000 so we can perform calculations with them later, but then there’s a risk that we’ll significantly distort the information
2) Keep the “<10,000” category, but then we can’t perform any calculations with these values
3) Replace all values with categories, sort of smoothing them out, but we can’t do that either, since populations in different countries may vary, and 10,000 people in a country with a population of one million and 40 million need to be coded into different categories

If we solve this problem, we will be able to explore many aspects related to the changing trends in the number of people belonging to a particular religion in a particular country